vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -382,7 +344,34 @@ protected <T> GridSelectionModel<T> createModel(Grid<T> grid) {
          *            the grid to create the selection model for
          * @return the selection model
          */
-        protected abstract <T> GridSelectionModel<T> createModel(Grid<T> grid);
+        protected <T> GridSelectionModel<T> createModel(Grid<T> grid) {
+            return switch (this) {
+            case SINGLE -> new AbstractGridSingleSelectionModel<T>(grid) {
```

Comment:
Just thinking out load: are you sure the oldish ASM can work with this newish switch? 😬

Reply:
Good catch! But believe or not, this new switch doesn't trigger any errors in Payara, application starts fine  :sweat_smile: don't ask me why :)